### PR TITLE
Dark mode toggling feature

### PR DESCRIPTION
**Pull request by swoocn**
I've explored implementing a dark mode feature, and while the current approach appears to be functional, I believe there might be more straightforward methods to integrate it within Angular. As we proceed with building additional screens, it might be worth revisiting this implementation. As of now, toggling dark mode will only affect the header.

### CODE REVIEW

File: frontend/src/app/core/service/dark-mode.service.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+/* the service provides a centralized and reusable way to manage dark mode functionality throughout the app. */
+export class DarkModeTogglerService {
+  private darkModeSubject: BehaviorSubject<boolean>;
+  public darkMode$: BehaviorSubject<boolean>;
```
Comment (zolcsi) [inline frontend/src/app/core/service/dark-mode.service.ts]: this should rather be an observable

File: frontend/src/app/core/service/dark-mode.service.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  constructor() {
+    this.darkModeSubject = new BehaviorSubject<boolean>(false);
+    this.darkMode$ = this.darkModeSubject;
```
Comment (zolcsi) [inline frontend/src/app/core/service/dark-mode.service.ts]: then like: this.darkMode$ = this.darkModeSubject.asObservable();

File: frontend/src/app/core/service/dark-mode.service.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  }
+
+  toggleDarkMode() {
+    const currentMode = this.darkModeSubject.value;
+    this.darkModeSubject.next(!currentMode);
+  }
+
+  getDarkMode(): boolean {
```
Comment (zolcsi) [inline frontend/src/app/core/service/dark-mode.service.ts]: this method we don't need I guess

File: frontend/src/app/shared/dark-mode-toggler/dark-mode-toggler.component.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  selector: 'app-dark-mode-toggler',
+  standalone: true,
+  imports: [MatButtonModule, MatIconModule],
+  templateUrl: './dark-mode-toggler.component.html',
+  styleUrl: './dark-mode-toggler.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+
+export class DarkModeTogglerComponent {
+  @HostBinding('class.dark-mode') isDarkMode = false;
```
Comment (zolcsi) [inline frontend/src/app/shared/dark-mode-toggler/dark-mode-toggler.component.ts:16]: do we need this code?

File: frontend/src/app/shared/dark-mode-toggler/dark-mode-toggler.component.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+  constructor(private darkModeService: DarkModeTogglerService) { }
+
+  ngOnInit(): void {
```
Comment (zolcsi) [inline frontend/src/app/shared/dark-mode-toggler/dark-mode-toggler.component.ts:20]: do we need this code? :)

File: frontend/src/app/shared/dark-mode-toggler/dark-mode-toggler.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { DarkModeTogglerComponent } from './dark-mode-toggler.component';
+import { DarkModeTogglerService } from '../../core/service/dark-mode.service';
+import { MatIconModule } from '@angular/material/icon';
+import { MatButtonModule } from '@angular/material/button';
+import { BehaviorSubject } from 'rxjs';
+
+describe('DarkModeTogglerComponent', () => {
```
Comment (zolcsi) [inline frontend/src/app/shared/dark-mode-toggler/dark-mode-toggler.component.spec.ts:7]: ❤️

File: frontend/src/styles.scss [hunk 7]
```diff
@@ -1,6 +1,26 @@
+@use "../node_modules/@angular/material/" as mat;
+
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
 
 html, body { height: 100%; }
 body { margin: 0; font-family: Roboto, "Helvetica Neue", sans-serif; min-width: 21rem;}
+
+.optional-controls { display:flex; align-items: center; }
```
Comment (zolcsi) [inline frontend/src/styles.scss]: where is this class used?

**Comment by zolcsi** [comment]
Great initiative! I think there is a need for dark mode. Indeed it's only on the header, but I guess the map is dark and day the same. But if we go in the 'add my business' it my be good to have a dark background with white text. Do you know how would be the best to apply that?

File: frontend/src/app/shared/dark-mode-toggler/dark-mode-toggler.component.ts (hunk 4 above)
Comment (zolcsi) [inline frontend/src/app/shared/dark-mode-toggler/dark-mode-toggler.component.ts]: we may need to set the class on the body, to have effect on the whole page, like here https://dev.to/omamaaslam/a-beginners-guide-to-adding-lightdark-mode-toggle-in-angular-material-47h7

File: frontend/src/app/core/service/dark-mode.service.ts (hunk 1 above)
Comment (swoocn) [reply frontend/src/app/core/service/dark-mode.service.ts]: I'm curious about the viability of using Observable in scenarios involving dialogs and popups, where it's crucial to know the current state of dark mode to apply the dark theme dynamically.  Wouldn't BehaviorSubject offer more flexibility for such scenarios?  Perhaps you know something that I don't..

File: frontend/src/styles.scss (hunk 7 above)
Comment (swoocn) [reply frontend/src/styles.scss]: This is used in the header-component.html to wrap with your language switcher feature in the mat-toolbar.

```
<div class="optional-controls">
    <app-language-switcher></app-language-switcher>
    <app-dark-mode-toggler></app-dark-mode-toggler>
  </div>
```

**Review by zolcsi — APPROVED** [review]
LGTM

**Pull request merged**